packages/sync/legacy/class.jetpack-sync-main.php

```diff
@@ -23,10 +37,23 @@ public static function init() {
 		 * Init after plugins loaded and before the `init` action. This helps with issues where plugins init
 		 * with a high priority or sites that use alternate cron.
 		 */
-		add_action( 'plugins_loaded', array( 'Jetpack_Sync_Actions', 'init' ), 90 );
+		add_action( 'plugins_loaded', array( $this, 'initialize_actions' ), 90 );
 
 		// We need to define this here so that it's hooked before `updating_jetpack_version` is called.
 		add_action( 'updating_jetpack_version', array( 'Jetpack_Sync_Actions', 'cleanup_on_upgrade' ), 10, 2 );
 		add_action( 'jetpack_user_authorized', array( 'Jetpack_Sync_Actions', 'do_initial_sync' ), 10, 0 );
 	}
+
+	public function initialize_actions() {
+		new Jetpack_Sync_Actions( $this->connection_manager );
+	}
+
+	/**
+	 * Set the connection manager object.
+	 *
+	 * @return Connection_Manager $connection_manager The connection manager object.
+	 */
+	public function get_connection_manager() {
```

Comment:
We won't need this method provided the external references to it (see `sync_allowed` comment, above) are moved into this class.

Reply:
Indeed